### ISSUE TITLE
findById should support paranoid = false option

### ISSUE THREAD

**Issue by bkoltai** [sequelize/sequelize#9626]
## What are you doing?

`findById` should allow a `paranoid` option to find soft-destroyed models.
```js
const Model = sequelize.define(
  "model",
  {
    name: DataTypes.STRING
  },
  {
    paranoid: true
  }
)
model = await Model.create({name: "foo"})
await model.destroy()
Model.findById(model.id, {paranoid: false})
```

## What do you expect to happen?
`findById` should support `paranoid = false`

## What is actually happening?
It doesn't.

__Dialect:__ any
__Sequelize version:__ 4.37.10
__Tested with latest release:__ No (If yes, specify that version)

I am mainly filing this for future seekers and to hear the reasoning why it isn't supported. If it can be supported, I may be able to submit a PR to add this.

Thanks for your review

**Comment by paulmowat** [comment]
I've just tried this on version 5.0.0-beta.9 with the below code.

```
var Sequelize = require('../sequelize')
var fs = require('fs')
var path = require('path')

var sequelize = new Sequelize('sequelizesandbox', 'sequelizesandbox', 'sequelizesandbox', {
  host: 'localhost',
  dialect: 'postgres'
})

var UserPrimary = sequelize.define('UserWithPrimaryKey', {
  id: {type: Sequelize.STRING, primaryKey: true},
  name: Sequelize.STRING}, 
  {paranoid: true}  
);

return UserPrimary.sync({ force: true }).then(() => {
  return Promise.all([
    UserPrimary.create({
    id: '1',
    name: 'Paul'
  }),
  UserPrimary.create({
    id: '2',
    name: 'Bob'
  })])
  .then(() => {
    return UserPrimary.destroy({where: {id: '1'}})
  })
  .then(() => {
    return UserPrimary.findById('1', {paranoid: true}).then(u2 => {
      console.log(JSON.stringify(u2, null, '\t'))
    });
  })
  .then(() => {
    return UserPrimary.findById('1', {paranoid: false}).then(u2 => {
      console.log(JSON.stringify(u2, null, '\t'))
    });
  })
  .then(() => {
    return UserPrimary.findById('2', {paranoid: true}).then(u2 => {
      console.log(JSON.stringify(u2, null, '\t'))
    });
  })
  .then(() => {
    return UserPrimary.findById('2', {paranoid: false}).then(u2 => {
      console.log(JSON.stringify(u2, null, '\t'))
    });
  });
})
.catch((err) => {
  console.log(err)
  process.exit(1)
});
```
For record 1 - Deleted

If paranoid = true, then I'm not seeing the record as it was soft deleted.

Returns 

null

If paranoid = false, then I am seeing the record. Along with the deletedAt date value populated.

Returns

{
	"id": "1",
	"name": "Paul",
	"createdAt": "2018-07-04T12:56:07.898Z",
	"updatedAt": "2018-07-04T12:56:07.898Z",
	"deletedAt": "2018-07-04T12:56:08.463Z"
}

For record 2 - Not Deleted

If paranoid = true or false then I am seeing the record as it has not been deleted. 

Returns 

{
	"id": "2",
	"name": "Bob",
	"createdAt": "2018-07-04T12:56:07.899Z",
	"updatedAt": "2018-07-04T12:56:07.899Z",
	"deletedAt": null
}

From what I can tell based on the above this is working as required.

**Comment by sushantdhiman** [comment]
With `v4` sequelize try to compare `deleted_at` timestamp with current datetime, so it is possible `findById` failed because of timing issue.

As @paulmowat pointed out this works in `v5` because this behaviour of `paranoid` option was fixed in `v5.0.0-beta` https://github.com/sequelize/sequelize/blob/master/docs/upgrade-to-v5.md#500-beta

**Comment by bkoltai** [comment]
Awesome, works great, thanks!
